packages/shopping/src/__tests__/unit/utils.authentication.unit.ts

```diff
@@ -29,63 +32,160 @@ describe('authentication services', () => {
     );
+    const credentials = {email: 'dom@example.com', password: 'p4ss'};
     return expect(
-      jwtService.getAccessTokenForUser({
-        email: 'unittest@loopback.io',
-        password: 'fake',
-      }),
+      userService.validateCredentials(credentials),
     ).to.be.rejectedWith(expectedError);
   });
 
-  it('decodeAccessToken decodes valid access token', async () => {
-    const token = await jwtService.getAccessTokenForUser({
-      email: 'unittest@loopback.io',
-      password: 'p4ssw0rd',
+  it('user service verifyCredentials() succeeds', async () => {
+    const {email} = newUser;
+    const credentials = {email, password: user.password};
+
+    const returnedUser = await userService.verifyCredentials(credentials);
+
+    // create a copy of returned user without password field
+    const returnedUserWithOutPassword = Object.assign({}, returnedUser, {
+      password: user.password,
     });
-    const expectedUser = getExpectedUser(newUser);
-    const currentUser = await jwtService.decodeAccessToken(token);
-    expect(currentUser).to.deepEqual(expectedUser);
+    delete returnedUserWithOutPassword.password;
+
+    // create a copy of expected user without password field
+    const expectedUserWithoutPassword = Object.assign({}, newUser);
+    delete expectedUserWithoutPassword.password;
+
+    expect(returnedUserWithOutPassword).to.deepEqual(
+      expectedUserWithoutPassword,
+    );
+  });
+
+  it('user service verifyCredentials() fails with user not found', async () => {
+    const credentials = {email: 'idontexist@example.com', password: 'p4ssw0rd'};
+
+    const expectedError = new HttpErrors.NotFound(
+      `User with email ${credentials.email} not found.`,
+    );
+
+    await expect(userService.verifyCredentials(credentials)).to.be.rejectedWith(
+      expectedError,
+    );
+  });
+
+  it('user service verifyCredentials() fails with incorrect credentials', async () => {
+    const {email} = newUser;
+    const credentials = {email, password: 'invalidp4ssw0rd'};
+    const expectedError = new HttpErrors.Unauthorized(
+      'The credentials are not correct.',
+    );
+
+    await expect(userService.verifyCredentials(credentials)).to.be.rejectedWith(
+      expectedError,
+    );
+  });
+
+  it('user service convertToUserProfile() succeeds', async () => {
+    const expectedUserProfile = {
+      id: newUser.id,
+      name: `${newUser.firstname} ${newUser.surname}`,
+    };
+    const userProfile = userService.convertToUserProfile(newUser);
+    expect(expectedUserProfile).to.deepEqual(userProfile);
   });
 
-  it('decodeAccessToken throws error for invalid accesstoken', async () => {
-    const token = 'fake';
-    const error = new JsonWebTokenError('jwt malformed');
-    return expect(jwtService.decodeAccessToken(token)).to.be.rejectedWith(
-      error,
+  it('user service convertToUserProfile() succeeds without optional fields : firstname, surname', async () => {
```

Comment:
`convertToUserProfile` is not an async function, so let's remove the `async` keyword from the `it` block here.